In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns

In [26]:
df = pd.read_csv('/Users/cesurdagli/Desktop/Capstone Project/Feb17NoNull_Merged.csv')
print(df.shape)


(10257, 53)


In [3]:
df=df.drop(['Unnamed: 0'], axis=1)

# Infrential Statistics 
    Are there variables that are particularly significant in terms of explaining the answer to your project question?


    Are there strong correlations between pairs of independent variables or between an independent and a dependent variable?

    What are the most appropriate tests to use to analyse these relationships?

In my project, the variables created to meet the theory of First Principles of Instruction are significant to check the relationship between those variables and student result. According to the theory, those who experience First Principles of Instruction are more likely to have better scores compared to those who don't. Therefore, I create a new column called fpi for those who stated they experience First Principles of Instruction by calculating each score of ['ave_authenticProblems', 'ave_activation', 'ave_demonstration', 'ave_application','ave_integration'] grater than 3.5 so that I can conclude that those with score grater than 3.5 experienced the First Principles of Instruction.

In [27]:
df['fpi']=['yes' if df['ave_activation'][b]>3.5 and df['ave_authenticProblems'][b]>3.5 and df['ave_demonstration'][b]>3.5 and df['ave_application'][b]>3.5 and df['ave_integration'][b]>3.5 else 'no' if df['ave_activation'][b]<=3.5 and df['ave_authenticProblems'][b]<=3.5 and df['ave_demonstration'][b]<=3.5 and df['ave_application'][b]<=3.5 and df['ave_integration'][b]<=3.5 else 'partially' for b in range(len(df))]


# Anova
Now, I would like to run some infrential statistics. First, I would like to run one-way ANOVA to compare means of a continuous variable in two or more independent comparison groups. The hypothesis are as follows

H0: There is no difference among students with confident, some, and little level knowledge in the GR group based on their test duration, number of right questions, satisfaction with the tutorial respectively.

H1: There is a significant difference among group means.


In [5]:
#Spliting the dataframe into two sub datasetes based on GR and UG level
gr_df=df[df['testLevel']=='GR']
print(gr_df.shape)
ug_df=df[df['testLevel']=='UG']
print(ug_df.shape)

(2869, 53)
(7388, 53)


In [6]:
from scipy import stats

# Run Anova tests for the GR group, first.
# Check whether or not any of the group means of test durations differ from one another in GR group
print(stats.f_oneway(gr_df[gr_df['level']=='confident']['testDuration'], gr_df[gr_df['level']=='some']['testDuration'],gr_df[gr_df['level']=='little']['testDuration']))

# Check whether or not any of the group means of right questions differ from one another in GR group
print(stats.f_oneway(gr_df[gr_df['level']=='confident']['right_question'], gr_df[gr_df['level']=='some']['right_question'],gr_df[gr_df['level']=='little']['right_question']))

# Check whether or not any of the group means of satisfaction differ from one another GR group
print(stats.f_oneway(gr_df[gr_df['level']=='confident']['ave_quality_Satisfaction'], gr_df[gr_df['level']=='some']['ave_quality_Satisfaction'],gr_df[gr_df['level']=='little']['ave_quality_Satisfaction']))


F_onewayResult(statistic=2.8115178020814375, pvalue=0.060285424876925564)
F_onewayResult(statistic=21.809049125370194, pvalue=4.002470700440738e-10)
F_onewayResult(statistic=2.621216016118097, pvalue=0.07289494126703458)


I reject H0 for the second test. I have statistically significant evidence at α=0.05 to show that there is a difference in number of right questions among the three level of knowledge. I failed to reject H0 for first one and third one so that I I do not have statistically significant evidence at α=0.05 to show that there is a difference in test durations and satisfaction with tutorial. However, we don’t know where the difference between groups is yet so that I calculated the multiple t-tests. The T tests below showed that there are significant differences in the number of right questions between students with different knowledge level.


In [7]:
ind_t_test=stats.ttest_ind(gr_df[gr_df['level']=='confident']['right_question'],gr_df[gr_df['level']=='some']['right_question'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 4.967
	p-value = 0.000


In [8]:
ind_t_test=stats.ttest_ind(gr_df[gr_df['level']=='confident']['right_question'],gr_df[gr_df['level']=='little']['right_question'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 4.994
	p-value = 0.000


In [9]:
ind_t_test=stats.ttest_ind(gr_df[gr_df['level']=='some']['right_question'],gr_df[gr_df['level']=='little']['right_question'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 3.676
	p-value = 0.000


In [10]:
# Below I wanted to check the average number of rigth questions for student with confident, some, and little knowledge in the GR group.
print(gr_df[gr_df['level']=='confident']['right_question'].mean())
print(gr_df[gr_df['level']=='some']['right_question'].mean())
print(gr_df[gr_df['level']=='little']['right_question'].mean())

7.216987865810135
6.848167539267016
5.53125


In [11]:
# Annova tests for the UG group
# Check whether or not any of the group means of test durations differ from one another based on knowledge level in UG group
print(stats.f_oneway(ug_df[ug_df['level']=='confident']['testDuration'], ug_df[ug_df['level']=='some']['testDuration'],ug_df[ug_df['level']=='little']['testDuration']))

print(stats.f_oneway(ug_df[ug_df['level']=='confident']['right_question'], ug_df[ug_df['level']=='some']['right_question'],ug_df[ug_df['level']=='little']['right_question']))

print(stats.f_oneway(ug_df[ug_df['level']=='confident']['ave_quality_Satisfaction'], ug_df[ug_df['level']=='some']['ave_quality_Satisfaction'],ug_df[ug_df['level']=='little']['ave_quality_Satisfaction']))


F_onewayResult(statistic=2.0349745984723016, pvalue=0.1307604599473568)
F_onewayResult(statistic=37.3129143070894, pvalue=7.590398905584018e-17)
F_onewayResult(statistic=30.59384823663756, pvalue=5.895422160775933e-14)


I reject H0 for the second and third test. I have statistically significant evidence at α=0.05 to show that there is a difference in number of right questions and satisfaction with the tutorial among students with the three level of knowledge. I failed to reject H0 for firstso that I I do not have statistically significant evidence at α=0.05 to show that there is a difference in test durations students with three level of konwledge.

In [12]:
ind_t_test=stats.ttest_ind(ug_df[ug_df['level']=='confident']['right_question'],ug_df[ug_df['level']=='some']['right_question'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 5.652
	p-value = 0.000


In [13]:
ind_t_test=stats.ttest_ind(ug_df[ug_df['level']=='confident']['right_question'],ug_df[ug_df['level']=='little']['right_question'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 7.261
	p-value = 0.000


In [14]:
ind_t_test=stats.ttest_ind(ug_df[ug_df['level']=='some']['right_question'],ug_df[ug_df['level']=='little']['right_question'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 5.651
	p-value = 0.000


In [15]:
# Below I wanted to check the average number of rigth questions for student with confident, some, and little knowledge in the UG group.
print(ug_df[ug_df['level']=='confident']['right_question'].mean())
print(ug_df[ug_df['level']=='some']['right_question'].mean())
print(ug_df[ug_df['level']=='little']['right_question'].mean())

6.887795275590551
6.612263033175355
5.576923076923077


In [16]:
ind_t_test=stats.ttest_ind(ug_df[ug_df['level']=='confident']['ave_quality_Satisfaction'],ug_df[ug_df['level']=='some']['ave_quality_Satisfaction'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 5.105
	p-value = 0.000


In [17]:
ind_t_test=stats.ttest_ind(ug_df[ug_df['level']=='confident']['ave_quality_Satisfaction'],ug_df[ug_df['level']=='little']['ave_quality_Satisfaction'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 6.369
	p-value = 0.000


In [18]:
ind_t_test=stats.ttest_ind(ug_df[ug_df['level']=='some']['ave_quality_Satisfaction'],ug_df[ug_df['level']=='little']['ave_quality_Satisfaction'])
print('The results of the independent t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.3f}'.format(ind_t_test[0],ind_t_test[1]))

The results of the independent t-test are: 
	t-value = 5.318
	p-value = 0.000


In [19]:
print(ug_df[ug_df['level']=='confident']['ave_quality_Satisfaction'].mean())
print(ug_df[ug_df['level']=='some']['ave_quality_Satisfaction'].mean())
print(ug_df[ug_df['level']=='little']['ave_quality_Satisfaction'].mean())

3.814187289088864
3.7217120853080567
3.371153846153846


## Chi Square Test

Now, I would also like to run Chi Square test to check the relationship between categroical variables. My hypothesis:

H0: There is no statistically significant relationship between test result and experiencing first principles of instruction.

H1: There is a statistically significant relationship between test results and  experiencing first principles of instruction.

The next step is to format the data into a frequency count table by using the pd.crosstab() function in pandas.

In [20]:
# Constructing the Contingency Table for the GR group to summarize the relationship 
#between several categorical variables.
contingency_table = pd.crosstab(
    gr_df['pass_fail'],
    gr_df['fpi'],
    margins = True
)
contingency_table

fpi         no  partially   yes   All
pass_fail                            
FAIL       143        701  1294  2138
PASS        12        210   509   731
All        155        911  1803  2869

In [21]:
f_obs = np.array([contingency_table.iloc[0][:].values,
                  contingency_table.iloc[1][:].values])
f_obs

array([[ 143,  701, 1294, 2138],
       [  12,  210,  509,  731]])

In [22]:
stats.chi2_contingency(f_obs)[0:3]

(35.699454712664846, 8.668097054230568e-08, 3)

The X2= ~36, p-value = ~0 and degrees of freedom = 3. As seen above, With a p-value < 0.05 , we can reject the null hypothesis. There is a statistically significant relationship between test results and First Principles of Instruction in the GR group. We don't know what this relationship is, but we do know that these two variables are not independent of each other.

As shown in contingency table, there were 1803 out of 2869 GR test takers who agreed with having experienced all of the first principles. Given that the antecedent condition was true, the consequent (student mastery is ‘Pass’), was true in 509 of the 1803 cases, which yields a probability estimate of 509/1803 or 0.282. Thus, 28.2% of students who were rated as high masters by a certification test reported that they also agreed that first principles were experienced when using the tutorial.

Next the pattern was investigated where the antecedent condition was no agreement with having experienced first principles and where the consequent was high mastery.  For this pattern, the antecedent occurred 155 times, the consequent occurred in 12 of those 155 cases, for a probability estimate of 12/155 = 0.07. Thus, 7% of students who demonstrated high mastery of the objectives of the tutorial failed to agree that first principles occurred.  

This finding can be further interpreted as follows: When students perceived that first principles occurred; they were more than four times as likely (0.282/0.07 = 4.02) to be high masters compared to those participants who did not agree that first principles were experienced in the tutorials.  
 

In [23]:
# Constructing the Contingency Table for the UG group
contingency_table1 = pd.crosstab(
    ug_df['pass_fail'],
    ug_df['fpi'],
    margins = True
)
contingency_table1

fpi         no  partially   yes   All
pass_fail                            
FAIL       906       2263  2627  5796
PASS        91        559   942  1592
All        997       2822  3569  7388

In [24]:
f_obs1 = np.array([contingency_table1.iloc[0][:].values,
                  contingency_table1.iloc[1][:].values])
f_obs1

array([[ 906, 2263, 2627, 5796],
       [  91,  559,  942, 1592]])

In [25]:
stats.chi2_contingency(f_obs1)[0:3]

(145.61173583275271, 2.32972136835883e-31, 3)

As seen above, With a p-value < 0.05 , we can reject the null hypothesis. There is a statistically significant relationship between test results and First Principles of Instruction in the UG group.

This can be also interpreted in this way: 

As shown in contingency table1, there were 3569 out of 7388 UG test takers who agreed with having experienced all of the first principles. Given that the antecedent condition was true, the consequent (student mastery is ‘Pass’), was true in 942 of the 3569 cases, which yields a probability estimate of 942/3569 or 0.263. Thus, 26.3% of students who were rated as high masters by a certification test reported that they also agreed that first principles were experienced when using the tutorial.

Next the pattern was investigated where the antecedent condition was no agreement with having experienced first principles and where the consequent was high mastery.  For this pattern, the antecedent occurred 997 times, the consequent occurred in 91 of those 997 cases, for a probability estimate of 91/997 = 0.09. Thus, 9% of students who demonstrated high mastery of the objectives of the tutorial failed to agree that first principles occurred.  

This finding can be further interpreted as follows: When students perceived that first principles occurred, **they were about three times as likely (0.263/0.09 = 2.92) to be high masters compared to those participants who did not agree that first principles were experienced in the tutorials.  


